api/features/tasks.py

```diff
@@ -1,17 +1,127 @@
+import json
+from datetime import timedelta
 from threading import Thread
+from typing import Optional, Union
 
-from environments.models import Webhook
-from features.models import FeatureState
+from django.db.models import Q
+from django.utils import timezone
+
+from environments.models import Environment, Webhook
+from features.models import (
+    Feature,
+    FeatureExport,
+    FeatureImport,
+    FeatureState,
+    FeatureStateValue,
+)
+from features.multivariate.models import MultivariateFeatureOption
+from features.value_types import BOOLEAN, INTEGER, STRING
+from features.versioning.versioning_service import get_environment_flags_list
+from projects.models import Project
+from task_processor.decorators import (
+    register_recurring_task,
+    register_task_handler,
+)
 from webhooks.constants import WEBHOOK_DATETIME_FORMAT
 from webhooks.webhooks import (
     WebhookEventType,
     call_environment_webhooks,
     call_organisation_webhooks,
 )
 
+from .constants import OVERWRITE, SKIP
 from .models import HistoricalFeatureState
 
 
+@register_recurring_task(
+    run_every=timedelta(hours=12),
+)
+def clear_stale_feature_imports_and_exports() -> None:
+    two_weeks_ago = timezone.now() - timedelta(days=14)
+    FeatureExport.objects.filter(created_date__lt=two_weeks_ago).delete()
+    FeatureImport.objects.filter(created_date__lt=two_weeks_ago).delete()
+
+
+@register_task_handler()
+def export_features_for_environment(
+    environment_id: int, tag_ids: Optional[list[int]] = None
+) -> None:
+    additional_filters = Q(
+        identity__isnull=True,
+        feature_segment__isnull=True,
+        feature_state_value__isnull=False,
+        feature__is_archived=False,
+    )
+
+    if tag_ids:
+        additional_filters &= Q(feature__tags__in=tag_ids)
+
+    environment = Environment.objects.get(id=environment_id)
+    feature_states = get_environment_flags_list(
+        environment=environment,
+        additional_filters=additional_filters,
+    )
+
+    payload = []
+    for feature_state in feature_states:
+        multivariate = []
+        for mv_option in feature_state.feature.multivariate_options.all():
+            mv_feature_state_value = (
+                feature_state.multivariate_feature_state_values.filter(
+                    multivariate_feature_option=mv_option
+                ).first()
+            )
+
+            multivariate.append(
+                {
+                    "percentage_allocation": mv_feature_state_value.percentage_allocation,
+                    "default_percentage_allocation": mv_option.default_percentage_allocation,
+                    "value": mv_option.value,
+                }
+            )
```

Comment:
Couple of comments here: 

1. Couldn't we do something more like: 

```python
multivariate = [
    {
        "percentage_allocation": mv_fsv.percentage_allocation,
        "default_percentage_allocation": mv_fsv.multivariate_option.default_percentage_allocation,
        "value": mv_fsv.multivariate_option.value
    } for mv_fsv in feature_state.multivariate_feature_state_values.select_related("multivariate_option").all()
]
```

2. Without having seen the import code yet, I'm a little unsure why we'd store the value as a single field rather than the strictly typed fields that exist on the original model. Feels like it just adds extra typing complexities on the import.

Reply:
1. I was a bit confused with the first point since it overlaps with a different class, but then I realized you meant `select_related("multivariate_feature_option")` in which case yes this works! Updated locally.

2. I'm adding typing to the export as a precaution since it is more robust that way.